tests/baselines/reference/arrayFilterBoolean.types

```diff
@@ -0,0 +1,16 @@
+=== tests/cases/compiler/arrayFilterBoolean.ts ===
+const mixed = [undefined, "string", null]
+>mixed : (string | null | undefined)[]
+>[undefined, "string", null] : (string | null | undefined)[]
+>undefined : undefined
+>"string" : "string"
+>null : null
+
+const result = mixed.filter(Boolean)
+>result : (string | null | undefined)[]

```

Comment:
Moving to draft as this doesn't seem to actually work (and its late for me)

Reply:
On my latest versions the readonly arrays work, but mutable arrays still aren't using the overload